src/openms/include/OpenMS/MATH/STATISTICS/KernelDensityEstimation.h

```diff
@@ -0,0 +1,239 @@
+
+    /**
+      @brief Fast kernel density estimation on a regular grid using FFT convolution.
+
+      Computes Gaussian kernel density estimates at equally-spaced grid points using
+      the FFT-based convolution algorithm. This approach scales as O(n + M*log(M))
+      compared to O(n*M) for direct evaluation, making it highly efficient for large
+      datasets or fine grids.
+
+      The algorithm:
+      1. Bins data onto a regular grid using linear binning
+      2. Computes FFT of binned data and Gaussian kernel
+      3. Performs pointwise multiplication in frequency domain
+      4. Inverse FFT to obtain density estimates
+
+      The grid extends from min(x) - cut*bw to max(x) + cut*bw.
+
+      Reference:
+      Silverman BW. (1982) "Algorithm AS 176: Kernel density estimation using the Fast
+      Fourier Transform." J. R. Statist. Soc. C 31(1):93-99. DOI: 10.2307/2347084
+
+      @param x Data values for which to estimate density
+      @param bw Bandwidth of the Gaussian kernel
+      @param gridsize Number of equally-spaced grid points (M). Should be power of 2 for FFT efficiency
+      @param cut Extension factor: grid extends cut*bw beyond data range on each side
+      @return Pair of vectors: (density_values, grid_points), each of length @p gridsize
+
+      @note Returns uniform density if data range is zero (all identical values)
+
+      @ingroup MathStatistics
+    */
+    OPENMS_DLLAPI std::pair<std::vector<double>, std::vector<double>> gridKdeFft(const std::vector<double>& x, 
+                                                                                   double bw, 
+                                                                                   std::size_t gridsize = 512, 
+                                                                                   double cut = 3.0);
+
+    /**
+      @brief Evaluate kernel density estimates at the data points themselves.
+
+      Computes KDE values at each input data point using the FFT-grid method followed
+      by cubic spline interpolation. This is more efficient than direct evaluation for
+      large datasets, as the FFT-grid computation scales as O(n + M*log(M)) followed by
+      O(n*log(M)) for spline interpolation, compared to O(n^2) for direct methods.
+
+      The function first computes KDE on a regular grid using grid_kde_fft(), then
+      interpolates these grid values to the query points using cubic spline interpolation.
+
+      @param x Data values at which to evaluate the density
+      @param bw Bandwidth of the Gaussian kernel
+      @param gridsize Number of grid points for FFT computation (default 512)
+      @param cut Extension factor for grid range (default 3.0)
+      @return Vector of KDE values, one for each point in @p x
+
+      @see gridKdeFft() for the underlying grid-based KDE computation
+
+      @ingroup MathStatistics
+    */
+    OPENMS_DLLAPI std::vector<double> kdeFftEval(const std::vector<double>& x, 
```

Comment:
maybe check once if we have other FFT functions (e.g. in evergreen) and if we expose names. e.g. if it could make sense to always write FFT instead of Fft for consistence.